tests/rptest/tests/prefix_truncate_recovery_test.py

```diff
@@ -69,7 +69,6 @@ def fully_replicated(self, nodes):
         metric = self.redpanda.metrics_sample("under_replicated_replicas",
                                               nodes)
         metric = metric.label_filter(dict(namespace="kafka", topic=self.topic))
-        assert len(metric.samples) == len(nodes)
         return all(map(lambda s: s.value == 0, metric.samples))
```

Comment:
technically all([]) is True,
since fully_replicated is called inside wait_until, maybe it should be 
```suggestion
        if len(metric.samples) != len(nodes):
            return False
```

Reply:
Thanks for catching this. I have updated the method and believe it catches all edge cases in a generic way now.